<?xml version="1.0" ?>
<?xml-model href="http://sdformat.org/schemas/root.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
<sdf version="1.5">
  <world name="default">
    <!-- <plugin name="mrs_gazebo_ros_link_attacher_plugin" filename="libmrs_gazebo_ros_link_attacher.so"/> -->
    <plugin name="mrs_gazebo_static_transform_republisher_plugin" filename="libMRSGazeboStaticTransformRepublisher.so"/>
      
    <!-- coordinate system {-->
    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <latitude_deg>37.411802</latitude_deg>
      <longitude_deg>-121.995739</longitude_deg>
      <elevation>0.0</elevation>
      <heading_deg>0</heading_deg>
    </spherical_coordinates>
    <!--}-->

    <!-- physics engine {-->
    <physics name="default_physics" default="0" type="ode">
      <gravity>0 0 -9.8066</gravity>
      <ode>
        <solver>
          <type>quick</type>
          <iters>10</iters>
          <sor>1.3</sor>
          <use_dynamic_moi_rescaling>0</use_dynamic_moi_rescaling>
        </solver>
        <constraints>z
          <cfm>0</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>1000</contact_max_correcting_vel>
          <contact_surface_layer>0.001</contact_surface_layer>
        </constraints>
      </ode>
      <max_step_size>0.004</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>250</real_time_update_rate>
      <magnetic_field>6.0e-06 2.3e-05 -4.2e-05</magnetic_field>
    </physics>
    <!--}-->

    <!-- turn off shadows {-->
    <scene>
      <shadows>false</shadows>
      <sky>
        <clouds/>
      </sky>
    </scene>
    <!--}-->

    <!-- sun {-->
    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>250 250 600 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>0 0 -1</direction>
    </light>
    <!--}-->

    <!-- ground plane {-->
    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <pose>0 0 0 0 0 0</pose>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>250 250</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>1</mu>
                <mu2>1</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="grass">
          <pose>0 0 0 0 0 0</pose>
          <cast_shadows>false</cast_shadows>
          <geometry>
            <mesh>
              <uri>file://grass_plane/meshes/grass_plane.dae</uri>
            </mesh>
          </geometry>
          <!-- <material> -->
          <!--   <script> -->
          <!--     <uri>file://media/materials/scripts/Gazebo.material</uri> -->
          <!--     <name>Gazebo/Grass</name> -->
          <!--   </script> -->
          <!-- </material> -->
        </visual>
      </link>
    </model>
    <!--}-->

<!-- 
    <model name='brick_box'>
      <link name='link_0'>
      <collision name='collision'>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>20 0.5 4</scale>
            </mesh>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
        </inertial>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>20 0.5 4</scale>
            </mesh>
          </geometry>
          <transparency>0</transparency>
          <cast_shadows>1</cast_shadows>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
      <pose frame=''>10 0 2 0 -0 0</pose>
    </model>

    <model name='brick_box_2'>
      <link name='link_1'>
      <collision name='collision'>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>20 0.5 4</scale>
            </mesh>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
        </inertial>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>20 0.5 4</scale>
            </mesh>
          </geometry>
          <transparency>0</transparency>
          <cast_shadows>1</cast_shadows>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
      <pose frame=''>10 10 2 0 -0 0</pose>
    </model>

    <model name='brick_box_3'>
      <link name='link_2'>
      <collision name='collision'>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>10 0.5 4</scale>
            </mesh>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
        </inertial>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>10 0.5 4</scale>
            </mesh>
          </geometry>
          <transparency>0</transparency>
          <cast_shadows>1</cast_shadows>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
      <pose frame=''>0 5 2 0 0 1.5707963267948966</pose>
    </model>

    <model name='brick_box_4'>
      <link name='link_3'>
      <collision name='collision'>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>10 0.5 4</scale>
            </mesh>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
        </inertial>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://Brick_Box/meshes/simple_box.dae</uri>
              <scale>10 0.5 4</scale>
            </mesh>
          </geometry>
          <transparency>0</transparency>
          <cast_shadows>1</cast_shadows>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
      <pose frame=''>20 5 2 0 0 1.5707963267948966</pose>
    </model> -->






































































































































































    
    <!-- THE VOID (for spawning objects out of camera view) {-->
    <model name='the_void'>
      <static>1</static>
      <link name='link'>
        <pose frame=''>0 0 0.1 0 -0 0</pose>
        <visual name='the_void'>
          <pose frame=''>0 0 2 0 -0 0</pose>
          <geometry>
            <sphere>
              <radius>0.25</radius>
            </sphere>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/Gazebo.material</uri>
              <name>Gazebo/Black</name>
            </script>
          </material>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <pose frame=''>-1000 -1000 0 0 0 0</pose>
    </model>
    <!--}-->

    <!-- user camera {-->
    <gui>
      <camera name="camera">
        <pose>-60 -100 30 0 0.4 0.89</pose>
      </camera>
    </gui>
    <!--}-->

    <!-- GUI frame synchronization {-->
    <plugin name="mrs_gazebo_rviz_cam_synchronizer" filename="libMRSGazeboRvizCameraSynchronizer.so" >
      <target_frame_id>gazebo_user_camera</target_frame_id>
      <world_origin_frame_id>uav1/gps_origin</world_origin_frame_id>
      <frame_to_follow>uav1::base_link</frame_to_follow>
    </plugin>
    <!--}-->

  </world>
</sdf>